packages/testing-lib/src/waitForState.ts

```diff
@@ -0,0 +1,19 @@
+import { act } from './RTLOverrides';
+
+/**
+ * Wrapper around `act`.
+ *
+ * Awaits an `act` call,
+ * and returns the value of the state update callback
+ */
+export const waitForState = async <T extends any>(
+  callback: () => T,
+): Promise<T> => {
+  let val: T;
+  await act(() => {
+    val = callback();
+  });
+
+  // @ts-expect-error
```

Comment:
What error are we expecting?

Reply:
Updated. `val` is technically being referenced before it's declared as far as the TS compiler is concerned